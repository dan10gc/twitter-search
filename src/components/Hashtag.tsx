import React from "react";

interface Props {
  hashtag: string;
}

const Hashtag = ({ hashtag }: Props) => {
  return <a className="badge">#{hashtag}</a>;
};

export default Hashtag;
