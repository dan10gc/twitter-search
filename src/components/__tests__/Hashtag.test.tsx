import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Hashtag from '../Hashtag';

const onHashtagPressMock = jest.fn();

describe('Hashtag Component', () => {
	test('renders hashtag', () => {
		const { container, getByText } = render(<Hashtag hashtag="reactjs" onHashtagPress={onHashtagPressMock} />);
		expect(getByText('#reactjs')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
