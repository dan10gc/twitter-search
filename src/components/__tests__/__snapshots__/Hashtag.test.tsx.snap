// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hashtag Component renders hashtag 1`] = `
<div>
  <div
    class="badge is-clickable"
  >
    <span>
       #
      reactjs
    </span>
  </div>
</div>
`;
